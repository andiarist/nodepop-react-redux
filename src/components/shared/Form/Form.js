import React, { useState } from 'react';
import T from 'prop-types';

import { FormContext } from './FormContext';

function Form({ initialValues, onSubmit, children }) {
  const [values, setValues] = useState(initialValues);
  const handleSubmit = ev => {
    ev.preventDefault();
    onSubmit(values);
  };
  return (
    <form onSubmit={handleSubmit}>
      <FormContext.Provider value={{ values, setValues }}>
        {children}
      </FormContext.Provider>
    </form>
  );
}

Form.propTypes = {
  onSubmit: T.func.isRequired,
  initialValues: T.object.isRequired,
  children: T.node.isRequired,
};

export default Form;
