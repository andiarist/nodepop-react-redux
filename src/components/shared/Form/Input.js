import React, { useContext } from 'react';
import T from 'prop-types';
import { FormContext } from './FormContext';

function Input({ type, name, ...props }) {
  const { values, setValues } = useContext(FormContext);

  const handleChange = event => {
    const valueChanging =
      event.target.type === 'checkbox'
        ? event.target.checked
        : event.target.value;
    setValues({ ...values, [name]: valueChanging });
  };

  return (
    <input type={type} name={name} onChange={handleChange} placeholder={name} />
  );
}
Input.propTypes = {
  type: T.string.isRequired,
  name: T.string.isRequired,
};

export default Input;
