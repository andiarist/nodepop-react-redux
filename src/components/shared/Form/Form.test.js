import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Testing <Form />', () => {
  test('should show <Form /> properly', () => {
    const props = {
      initialValues: {},
      onSubmit: jest.fn(),
      children: 'children',
    };
    const wrapper = shallow(<Form {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
