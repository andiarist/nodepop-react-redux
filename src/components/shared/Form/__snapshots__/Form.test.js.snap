// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Form /> should show <Form /> properly 1`] = `
<form
  onSubmit={[Function]}
>
  <ContextProvider
    value={
      Object {
        "setValues": [Function],
        "values": Object {},
      }
    }
  >
    children
  </ContextProvider>
</form>
`;
